### PR TITLE
Use reflection to call AsyncCausalityTracer and Task internal methods

### PR DESCRIPTION
**Pull request by luqunl**
Use reflection instead of InternalsVisibleTo to access internal methods inside System.Private.Corelib.

### CODE REVIEW

**Comment by jkotas** [comment]
Yes, I believe that was the plan we agreed over email.

File: src/System.Runtime.WindowsRuntime/src/System/Threading/Tasks/AsyncInfoToTaskBridge.CoreCLR.cs [hunk 1]
```diff
@@ -252,5 +241,140 @@ private void Complete(IAsyncInfo asyncInfo, Func<IAsyncInfo, TResult> getResults
                     Marshal.ReleaseComObject(asyncInfo);
             }
         }  // private void Complete(..)
+
+        private static Type s_asyncCausalityTracerType;
+        private static PropertyInfo s_LoggingOn;
+        private static MethodInfo s_TraceOperationCreationMethodInfo;
+        private static MethodInfo s_TraceOperationCompletionMethodInfo; 
+        private bool IsAsyncCausalityTracerLoggingOn()
```
Comment (jkotas) [inline src/System.Runtime.WindowsRuntime/src/System/Threading/Tasks/AsyncInfoToTaskBridge.CoreCLR.cs]: Note that this property is expected to be fast. I do not think that the use of reflection the way you have it here would fly well.

**Comment by luqunl** [comment]
@weshaggard/@jkotas,  working on "keep this as static references and just expose these as public in CoreLib"

**Comment by luqunl** [comment]
I will close this PR and open another PR for new solution.

**Pull request closed without merging**